scripts/llnl/common_build_functions.py

```diff
@@ -25,34 +25,42 @@
 from os.path import join as pjoin
 
 def sexe(cmd,
-         ret_output=False,
+         ret_output = False,
          output_file = None,
          echo = False,
          error_prefix = "ERROR:"):
     """ Helper for executing shell commands. """
     if echo:
         print("[exe: %s]" % cmd)
     if ret_output:
+        # Return exit code and output
         p = subprocess.Popen(cmd,
                              shell=True,
                              stdout=subprocess.PIPE,
                              stderr=subprocess.STDOUT)
-        res =p.communicate()[0]
+        res = p.communicate()[0]
         if isinstance(res, bytes):
             res = res.decode()
         return p.returncode,res
     elif output_file != None:
-        ofile = open(output_file,"w")
+        # Write to stdout and output file
         p = subprocess.Popen(cmd,
                              shell=True,
-                             stdout= ofile,
+                             stdout=subprocess.PIPE,
                              stderr=subprocess.STDOUT)
```

Comment:
Can you verify that stderr is going to both the file and the screen?

Reply:
Yes, it does. I created a small script as a test:

```python
sexe("echo hello!", output_file="log0.txt")
sexe("echo bye!", output_file="log1.txt")
sexe("cho oops!", output_file="log2.txt")
```

```bash
[meemee@ruby963:repo]$ ../shell_exec.py 
hello!
bye!
/bin/sh: cho: command not found
[meemee@ruby963:repo]$ cat log?.txt 
hello!
bye!
/bin/sh: cho: command not found
```